### DISCUSSION TITLE
api for soft/hard navigation control

### DISCUSSION THREAD

**Discussion by darklight9811** [vercel/next.js#53404] [Ideas]
### Goals

1. give power to the developer to decide if a route should be soft/hard loaded

### Non-Goals

1. Whether a page should be soft/hard loaded, next already handles this, we just want to be able to override it

### Background

For example in an app that I work on, when we deleted an item in its page, we would have to redirect to the index(list) of items of that type. But since its soft navigated, the deleted item shows there.

### Link
``` jsx
// type navigation = "soft" | "hard" | "auto"
<Link href="/posts navigation="soft" />
```

### router
``` jsx
const router = useRouter()
// type navigation = "soft" | "hard" | "auto"
router.push("/posts", { navigation: "hard" }) 
```

**Comment by davscro** [comment]
```
// hard
<a href="/posts/>
```

```
// hard
location.href = "/posts/"
```
You can always use plain implementations for hard linking  ;-)
